graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -147,55 +147,64 @@ class InputTypeGenerator(config: CodeGenConfig, document: Document) : BaseDataTy
-                        } + "}}"
-                    )
+        return checkAndGetLocaleCodeBlock(value, type)
+            ?: checkAndGetBigDecimalCodeBlock(value, type)
+            ?: when (value) {
+                is BooleanValue -> CodeBlock.of("\$L", value.isValue)
+                is IntValue -> CodeBlock.of("\$L", value.value)
+                is StringValue -> CodeBlock.of("\$S", value.value)
+                is FloatValue -> CodeBlock.of("\$L", value.value)
+                is EnumValue -> CodeBlock.of("\$T.\$N", type, value.name)
+                is ArrayValue -> if (value.values.isEmpty()) CodeBlock.of("java.util.Collections.emptyList()")
+                else CodeBlock.of(
+                    "java.util.Arrays.asList(\$L)",
+                    value.values.joinToString { v ->
+                        generateCode(v, type.className, inputTypeDefinitions).toString()
+                    }
+                )
+                is ObjectValue -> {
+                    val inputObjectDefinition = inputTypeDefinitions.first {
+                        val expectedCanonicalClassName = config.typeMapping[it.name] ?: "${config.packageNameTypes}.${it.name}"
+                        expectedCanonicalClassName == type.className.canonicalName()
+                    }
+                    if (value.objectFields.isEmpty()) {
+                        return CodeBlock.of("new $type()")
+                    } else {
+                        CodeBlock.of(
+                            "new $type(){{" + value.objectFields.joinToString("") { objectProperty ->
+                                val argumentType =
+                                    checkNotNull(inputObjectDefinition.inputValueDefinitions.find { it.name == objectProperty.name }) {
+                                        "Property \"${objectProperty.name}\" does not exist in input type \"${inputObjectDefinition.name}\""
+                                    }
+                                val argumentValue = generateCode(
+                                    objectProperty.value,
+                                    typeUtils.findReturnType(argumentType.type),
+                                    inputTypeDefinitions
+                                )
+                                "set${objectProperty.name.replaceFirstChar { it.uppercaseChar() }}($argumentValue);"
+                            } + "}}"
+                        )
+                    }
                 }
+                else -> CodeBlock.of("\$L", value)
             }
-            else -> CodeBlock.of("\$L", value)
-        }
     }
 
-    private fun checkAndGetLocaleValue(value: StringValue, type: JavaTypeName): String? {
-        if (type.toString() == "java.util.Locale") return "Locale.forLanguageTag(\"${value.value}\")"
-        return null
+    private fun checkAndGetLocaleCodeBlock(value: Value<out Value<*>>, type: JavaTypeName): CodeBlock? {
```

Comment:
This change seems mostly unrelated. I assume it prevents a crash in the event a non-string is passed in? Would be good to add a test case for.

Reply:
Also, assuming it does prevent a crash, perhaps rather than returning `null` it'd be better to throw some sort of exception with a friendly error message. If someone has set `someLocale: Locale = 123` in their schema, it's probably best we let them know rather than just silently not do anything.

Reply:
I think you're correct. Will do the changes.